numpy/distutils/tests/test_system_info.py

```diff
@@ -174,35 +175,37 @@ def test_temp2(self):
     def test_compile1(self):
         # Compile source and link the first source
         c = ccompiler.new_compiler()
+        previousDir = os.getcwd()
         try:
             # Change directory to not screw up directories
-            previousDir = os.getcwd()
             os.chdir(self._dir1)
             c.compile([os.path.basename(self._src1)], output_dir=self._dir1)
             # Ensure that the object exists
             assert_(os.path.isfile(self._src1.replace('.c', '.o')) or
                     os.path.isfile(self._src1.replace('.c', '.obj')))
-            os.chdir(previousDir)
-        except OSError:
+        except (OSError, CompileError):
```

Comment:
Although - now I think about it - I am not sure what error this test is checking for.